roles/pulp_repos/tasks/main.yml

```diff
@@ -0,0 +1,179 @@
+- name: Enable the CentOS PowerTools repo
+  block:
+
+    - name: Determine which file in /etc/yum.repos.d/ has the CentOS 8.0 to 8.2 PowerTools repo
+      shell: grep -l -E "^\[PowerTools\]" /etc/yum.repos.d/*.repo
+      register: repo_file
+      changed_when: false
+      failed_when: false
+      check_mode: False
+
+    - name: Enable the CentOS PowerTools repo
+      ini_file:
+        path: "{{ repo_file.stdout }}"
+        section: "PowerTools"
+        option: enabled
+        value: "1"
+        no_extra_spaces: true
+        mode: '0644'
+      when: repo_file.rc == 0
+      become: true
+
+    - name: Determine which file in /etc/yum.repos.d/ has the CentOS 8.3+ or Stream powertools repo
+      shell: grep -l -E "^\[powertools\]" /etc/yum.repos.d/*.repo
+      register: repo_file
+      changed_when: false
+      failed_when: false
+      check_mode: False
+
+    - name: Enable the CentOS powertools repo
+      ini_file:
+        path: "{{ repo_file.stdout }}"
+        section: "powertools"
+        option: enabled
+        value: 1
+        no_extra_spaces: true
+        mode: '0644'
+      when: repo_file.rc == 0
+      become: true
+
+  when:
+    - ansible_facts.distribution == "CentOS"
+    - ansible_facts.distribution_major_version|int >= 8
+    - __pulp_repos_centos_powertools_repo_enable_default and pulp_repos_centos_powertools_repo_requeted
+
+- name: Enable SCL repository
+  block:
+    # Try multiple possible names for the rhel7 SCL repo until it is found.
+    # The query ensures that a single string rather than a list of strings is valid.
+    - name: Find and enable the first found name of the RHEL7 SCL repo
+      include_tasks: ambiguously-named-repo.yml
+      loop: "{{ pulp_rhel7_scl_repo }}"
+      loop_control:
+        loop_var: __ambiguously_named_repo
+      vars:
+        reg_variable: "__pulp_database_repo_enabled"
+      when:
+        - ansible_facts.distribution == "RedHat"
+        - ansible_facts.distribution_major_version|int == 7
+        # Works for both strings and lists to make sure not empty
+        - pulp_rhel7_scl_repo is not none
+        - pulp_rhel7_scl_repo | length > 0
+        # Prevents running again once completed. Specific to the role name.
+        - __pulp_database_repo_enabled is not defined
+
+    - name: Install SCL repo packages on CentOS 7
+      package:
+        name:
+          - 'centos-release-scl-rh'
+          - 'centos-release-scl'
+        state: present
+      become: yes
+      when:
+        - ansible_facts.distribution == "CentOS"
+        - ansible_facts.distribution_major_version|int == 7
+  when: __pulp_rhel_scl_repo_enable_default and pulp_rhel_scl_repo_requested
+
+- name: "Configure {{ __pulp_pkg_repo_name }} RPM repository"
+  template:
+    src: pulpcore.repo.j2
+    dest: /etc/yum.repos.d/pulpcore.repo
+    mode: '0644'
+
+  when:
+    - ansible_facts.os_family == "RedHat"
+    - pulp_pkg_repo is not none
+  become: true
```

Comment:
Sigh, this also needs to be reworked in a similar way.

Only the pulp_common role needs it.